src/server/pachyderm_test.go

```diff
@@ -600,7 +600,78 @@ func TestMultipleInputsFromTheSameRepoDifferentBranches(t *testing.T) {
 	require.NoError(t, c.GetFile(commits[0].Commit.Repo.Name, commits[0].Commit.ID, "file", 0, 0, &buffer))
 	require.Equal(t, "data A\ndata B\n", buffer.String())
 }
+func TestRunPipeline(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+
+	c := getPachClient(t)
+	require.NoError(t, c.DeleteAll())
+
+	dataRepo := tu.UniqueString("TestRunPipeline_data")
+	require.NoError(t, c.CreateRepo(dataRepo))
+
+	branchA := "branchA"
+	branchB := "branchB"
+
+	pipeline := tu.UniqueString("pipeline")
+	require.NoError(t, c.CreatePipeline(
+		pipeline,
+		"",
+		[]string{"bash"},
+		[]string{
+			"cat /pfs/branch-a/file >> /pfs/out/file",
+			"cat /pfs/branch-b/file >> /pfs/out/file",
+			"echo ran-pipeline",
+		},
+		nil,
+		client.NewCrossInput(
+			client.NewPFSInputOpts("branch-a", dataRepo, branchA, "/*", false),
+			client.NewPFSInputOpts("branch-b", dataRepo, branchB, "/*", false),
+		),
+		"",
+		false,
+	))
 
+	commitA, err := c.StartCommit(dataRepo, branchA)
+	require.NoError(t, err)
+	c.PutFile(dataRepo, commitA.ID, "/file", strings.NewReader("data A\n"))
+	c.FinishCommit(dataRepo, commitA.ID)
+
+	commitB, err := c.StartCommit(dataRepo, branchB)
+	require.NoError(t, err)
+	c.PutFile(dataRepo, commitB.ID, "/file", strings.NewReader("data B\n"))
+	c.FinishCommit(dataRepo, commitB.ID)
+
+	iter, err := c.FlushCommit([]*pfs.Commit{commitA, commitB}, nil)
+	require.NoError(t, err)
+	commits := collectCommitInfos(t, iter)
+	require.Equal(t, 1, len(commits))
+	buffer := bytes.Buffer{}
+	require.NoError(t, c.GetFile(commits[0].Commit.Repo.Name, commits[0].Commit.ID, "file", 0, 0, &buffer))
+	require.Equal(t, "data A\ndata B\n", buffer.String())
+
+	commitM, err := c.StartCommit(dataRepo, "master")
+	require.NoError(t, err)
+	err = c.FinishCommit(dataRepo, commitM.ID)
+	require.NoError(t, err)
+
+	// we should have two jobs
+	ji, err := c.ListJob(pipeline, nil, nil)
+	require.NoError(t, err)
+	require.Equal(t, 2, len(ji))
+	// now run the pipeline
+	require.NoError(t, c.RunPipeline(pipeline, nil))
+	// running the pipeline should create a new job
+	require.NoError(t, backoff.Retry(func() error {
+		jobInfos, err := c.ListJob(pipeline, nil, nil)
+		require.NoError(t, err)
+		if len(jobInfos) != 3 {
+			return fmt.Errorf("expected 3 jobs, got %d", len(jobInfos))
+		}
+		return nil
+	}, backoff.NewTestingBackOff()))
+}
```

Comment:
I think some negative test cases would be good here as well, for example what happens if you pass unrelated branches, that aren't part of any input, to `RunPipeline`.

Reply:
done